python/ray/dag/collective_node.py

```diff
@@ -0,0 +1,197 @@
+from weakref import ReferenceType
+from typing import Any, Dict, List, Union, Tuple, Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import torch
+
+import ray
+from ray.dag import (
+    DAGNode,
+    ClassNode,
+)
+from ray.dag.constants import (
+    PARENT_CLASS_NODE_KEY,
+    BIND_INDEX_KEY,
+    COLLECTIVE_GROUP_KEY,
+)
+from ray.dag.format_utils import get_dag_node_str
+from ray.util.annotations import DeveloperAPI
+from ray.util.collective.nccl_types import CollectiveOp, ReduceOp
+from ray.experimental.channel import ChannelContext
+from ray.experimental.channel.torch_tensor_nccl_channel import _init_nccl_group
+from ray.experimental.channel.torch_tensor_type import GPUCommunicator, TorchTensorType
+
+
+class CollectiveGroup:
+    """Represent metadata for a NCCL collective method."""
```

Comment:
NCCL collective operation (instead of method)?

Can you expand the doc more? Mention the input_nodes, constraints etc.

Reply:
Done.